lib/vrl/stdlib/src/parse_grok.rs

```diff
@@ -110,66 +66,95 @@ impl Function for ParseGrok {
             .try_bytes_utf8_lossy()
             .expect("grok pattern not bytes")
             .into_owned();
-
-        let mut grok = grok::Grok::with_default_patterns();
-        let pattern =
-            Arc::new(grok.compile(&pattern, true).map_err(|e| {
-                Box::new(Error::InvalidGrokPattern(e)) as Box<dyn DiagnosticMessage>
-            })?);
-
-        Ok(Box::new(ParseGrokFn { value, pattern }))
+        let patterns = [pattern];
```

Comment:
Would it be possible/reasonable to reuse the entire `ParseGroks::compile` function? (wrapping the pattern in an `Expr::Literal` that returns a single element array).

Reply:
@fuchsnj patching up the compiler to being able to rewrite expressions in function arguments felt a little too hacky.

However, I made `parse_grok` and `parse_groks` now share most of the relevant parts in `compile`/`compile_arguments`: 190cabde8637a5de7ec851b0bc4e4dbb620aba8e

Reply:
I thought that's what might happen. Was hoping it would have been easier to directly delegate to `parse_groks`. Maybe one day.